sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1767,8 +1786,32 @@ def _more_tags(self):
             }
         }
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.4
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = (
+            MetadataRouter(owner=self.__class__.__name__)
+            .add_self_request(self)
+            .add(
+                splitter=self.cv,
```

Comment:
`cv` can be an int, we need:

```suggestion
                splitter=check_cv(self.cv),
```

Reply:
By the way should this also be done in BaseSearchCV and LogisticRegressionCV as well? Currently both use self.cv in metadata routing whereas in the code where they are used check_cv is used.

Reply:
Ideally yes, but I think it works now since `get_routing_for_object` returns an empty routing object for ints.